### DISCUSSION TITLE
Error: ENOENT: no such file or directory, lstat '/vercel/path0/frontend/.next/export-detail.json'

### DISCUSSION THREAD

**Discussion by hanpham32** [vercel/community#6575] [Help]
### Summary
```
Collecting build traces ...
Route (app)                              Size     First Load JS
┌ λ /                                    3.95 kB        88.7 kB
├ λ /_not-found                          887 B          85.6 kB
└ λ /dashboard                           112 kB          207 kB
+ First Load JS shared by all            84.8 kB
  ├ chunks/171-38f47a0fee7282cf.js       29.2 kB
  ├ chunks/1b8a090b-a4ac2385caa5125b.js  53.4 kB
  └ other shared chunks (total)          2.19 kB
λ  (Dynamic)  server-rendered on demand using Node.js
Traced Next.js server files in: 57.494ms
Error: ENOENT: no such file or directory, lstat '/vercel/path0/frontend/.next/export-detail.json'
```
### Screenshot
<img width="1165" alt="Screenshot 2024-04-21 at 11 34 41 AM" src="https://github.com/vercel/community/assets/36429830/01e70426-c318-4218-8fb3-8f9ee1c11048">

### Info
I'm using Next.js version 14.1.4

### Next config
```
next.config.mjs
/** @type {import('next').NextConfig} */
const nextConfig = {
    reactStrictMode: true,
    swcMinify: false,
};

export default nextConfig;
```

### The Bug
`pnpm run build` works fine on local machine, but when I run `vercel build` I encounter the error.
`Error: ENOENT: no such file or directory, lstat '/Users/hanpham/Developer/Hackathons/daily-enigma/frontend/.next/export-detail.json'`

### Expected Behavior
Deployment should be successful as local build is fine.

### Steps to Reproduce
`vercel build`

**Comment by versecafe** [comment]
@hanpham32 It's hard to say without your repo but if you're using turborepo it doesn't track cache changes in json files by default so making, or renaming a json won't trigger a rebuild, this is usually an issue for Nextra projects but can occur in other places, if it's not that it may be you have an untracked file and it's not actually in production so just double check that you can find the specified json in `https://vercel.com/[team]/[project]/[deployment]/source`

**Reply by hanpham32** [reply]
Hi @versecafe, here's my repo: https://github.com/hanpham32/daily-enigma

**Reply by hanpham32** [reply]
I think `export-detail.json` only gets generated by build command; it is not something I add in.

**Reply by versecafe** [reply]
@hanpham32 `export-detail.json` should only exist if your next config is set to `output: "export"` which yours is not so I don't see why it would be generated or needed, and on a side note you have some unneeded flags for the experimental appDir, it's stable and supported by default, a cleaned version of your config should look like this

```mjs
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: false,
};

export default nextConfig;
```

**Reply by hanpham32** [reply]
@versecafe I previously modified my next config based on internet examples in attempt to debug. I now have updated my next config same to what you have, tho I still face the same issue
`Error: ENOENT: no such file or directory, lstat '/vercel/path0/frontend/.next/export-detail.json'`

Do you know what might be causing this? I've tried doing internet searches tho I don't see anyone has similar problem and has the solution.

**Reply by hanpham32** [reply]
I don't know why my Vercel is checking for the file `export-detail.json` when it's not needed. I've created a new Vercel project and deployed again but still has same problem

**Reply by versecafe** [reply]
I'm not sure I can try debugging on Vercel itself if you give me some filler env vars to test with, if they're sensitive I can share my email so you don't have to post them, but this is certainly weird it really should not be trying to check that file.

**Reply by hanpham32** [reply]
Yes, the env vars are sensitive. Can I share them with you via email?

**Reply by versecafe** [reply]
sent you an email, hopefully I can resolve this soon

**Comment by versecafe** [comment]
Would like to make a note to anyone who stumbles onto this now that it has been fixed, certain libraries when used on the client side can for some reason trigger Vercel to expect an `export-detail.json` without the Next config being configured for exports, the issue resolves itself when moved to server side code but this is still unexpected behaviour, It is caused by one of the following libraries

```
"@web3-storage/w3up-client";
"@web3-storage/w3up-client/stores/memory";
"@ucanto/core/delegation";
"@ipld/car";
"@ucanto/principal/ed25519";
"@web3-storage/w3up-client/types";
```

@amyegan I'm not sure where to report this but there is definitely an issue with this in Vercel's deployment pipeline that this is triggering as it works fine in local production on macOS and in an Ubuntu VM, I can share all the logs from it.

**Reply by joshpitkin** [reply]
Any workaround for this?  I can confirm the same if the above libraries are imported and used to generate a [w3 client](https://web3.storage/docs/how-to/upload/#using-the-js-client) in a `app/api/*/route.ts` file which should be server side code..

**Comment by amyegan** [comment]
Hey @hanpham32. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use without sensitive environment variables?

@versecafe Thanks for your help investigating this

**Reply by hanpham32** [reply]
Hi @amyegan, here is the reproducible example repo I created: https://github.com/hanpham32/vercel-reproducible-example

I followed the [guideline](https://vercel.com/guides/creating-a-minimal-reproducible-example) adhered to reproducible example. I also generated some of the env vars and made them public so your team can have access to work with. The shared repo shows same issue I faced in my project.

**Reply by amyegan** [reply]
Thanks for sharing an example, @hanpham32. I'm not able to run the repo you provided because of environment variables are required. Can you remove the parts that need env vars to share the smallest possible repro example?

**Reply by versecafe** [reply]
It does need the env vars for the libraries that trigger the issue to work otherwise the libs throw an error over invalid env vars

**Reply by amyegan** [reply]
Right, so I would need an example that doesn't require valid env vars. Otherwise I can't use the example to see the error happening.

If the error only happens with packages that need to contact external services, then it sounds like the problem may be caused by something attempting to write to a read-only directory and not finding the file there later. Do you see the same error if you [deploy from a local build](https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt) using `vercel build` and `vercel deploy --prebuilt`?

**Reply by joshpitkin** [reply]
Here is another minimal reproducable example: https://github.com/joshpitkin/vercel-6575  This is just a basic `create-next-app` with an added api/test/route.ts file which imports and runs the functions to setup a w3up-client.  I added fake strings where normally environment variables would be used, but that should't matter for the `vercel build` to run..

I get the error with `vercel build` while `npm run build` works fine.  The error goes away if I comment out this [line](https://github.com/joshpitkin/vercel-6575/blob/main/app/api/test/route.ts#L15)

Error message:
`Error: ENOENT: no such file or directory, lstat '.../vercel-issue-6575/.next/export-detail.json'`
